pulsar-common/src/main/java/com/yahoo/pulsar/common/api/Commands.java

```diff
@@ -429,11 +456,36 @@ private static ByteBuf serializeCommandSendWithSize(BaseCommand.Builder cmdBuild
             cmd.writeTo(outStream);
             cmd.recycle();
             cmdBuilder.recycle();
+            
+            //Write checksum
+            if (includeChecksum) {
+                headers.writeShort(magicCrc32c);
+                checksumReaderIndex = headers.writerIndex();
+                headers.writeZero(4); // write dummy checksum int to skip 4 bytes in write index
+            }
 
+            int metadataReaderIndex = headers.writerIndex();
             // Write metadata
             headers.writeInt(msgMetadataSize);
             msgMetadata.writeTo(outStream);
 
+            if (includeChecksum) {
+                // mark reader and write index
+                int headersReaderIndex = headers.readerIndex();
+
+                // compute checksum
+                headers.retain();
+                payload.retain();
+                headers.readerIndex(metadataReaderIndex);
+                ByteBuf msgMetadataBuf = DoubleByteBuf.get(headers, payload);
```

Comment:
After #44 gets merged I guess we should make it work with incremental checksum for all kinds of buffers